### ISSUE TITLE
Embedded C++ breaks syntax highlighting.

### ISSUE THREAD

**Issue by twadleigh** [julia-vscode/julia-vscode#765]
I have captured an example of the failure in https://github.com/twadleigh/julia-syntax-test-cases/blob/master/embedded_code.jl. From line 43 on the code is highlighted as though it is mostly a C++ string.

For the record, https://github.com/twadleigh/julia-syntax-test-cases/blob/master/corner_cases.jl also shows another leaky failure.

**Comment by tshort** [comment]
Does it work okay in Atom?

It looks okay in Github's rendering which also uses Atom's grammar via Linguist.

**Comment by twadleigh** [comment]
Yes, I have used `Cxx.jl` with embedded C++ pretty regularly and without issue in Juno/Atom for about 1.5 yrs. I'm trying to move more of my workflow to vscode, and, given how much of my julia work involves `Cxx.jl` (most of it), this is blocker for me.

I was able to trace the failure to an overly aggressive "improvement" of `source.cpp` that is used by vscode. It's author claims it is based on the Atom grammar, but adds some structure at the top that ultimately breaks things for `julia-vscode`. Thankfully, that same unifying structure makes the workaround pretty easy, too.

I have already done the bulk of the work for a PR that fixes this one issue, though the test cases I refer to above still exhibit other breakage.

The primary piece of that PR is a julia script that automatically downloads the `julia.cson` from `atom-language-julia`, converts it to `json`, and applies a few substitutions to make it vscode-friendly. Expect to see it before the end of the day.
